src/util/macro_util.rs

```diff
@@ -393,6 +393,12 @@ macro_rules! repr_c_struct_has_padding {
         );
         layout.requires_static_padding() || layout.requires_dynamic_padding()
     }};
+    (@field ([$t:ty])) => {
+        <[$t] as $crate::KnownLayout>::LAYOUT
+    };
+    (@field ($t:ty)) => {
+        $crate::DstLayout::for_unpadded_type::<$t>()
+    };
```

Comment:
@jswrenn lmk if you can think of something here – it'd be great to be able to *merge* these with the following branches rather than just *add* them so we have branch duplication. I tried to move the parentheses up to the top branch's match, but that didn't work (that said, I didn't try that hard to get it to work). This is very minor, so we can just leave it, but it'd be nice to merge these if it turns out there's something easy we can do.